manifests/app.pp

```diff
@@ -0,0 +1,73 @@
+define  parse_platform::app (
+
+  $app_name       = $name,
+  $application_id = undef,
+  $master_key     = undef,
+  $port           = 1337,
+  $parse_root     = "/srv",
+  $cloud_code     = false
+
+) {
+
+  if $cloud_code {
+    warning("cloud_code enabled.")
+    warning("Put your code on $ ${parse_root }/${app_name}/cloud.")
+  }
+
+  include parse_platform
+
+  validate_string($application_id)
+
+  if $master_key == undef {
+    fail('master_key must be a string')
+  }
+
+  validate_integer($port)
+
+  file { "parse_root_${app_name}":
+    ensure => directory,
+    path   => "${parse_root}/${app_name}",
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0755',
+  }
+
+  file { "parse_cloud_${app_name}":
+    ensure => directory,
+    path   => "${parse_root}/${app_name}/cloud",
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0755',
+    require => File["parse_root_${app_name}"]
+  }
+   file{"parse_main_${app_name}":
```

Comment:
Wrong indentation. https://docs.puppet.com/guides/style_guide.html#spacing-indentation-and-whitespace